spec/lib/extensions/database_configuration_spec.rb

```diff
@@ -11,14 +13,20 @@
 
   context "ERB in the template" do
     before do
-      database_config = StringIO.new "---\nvalue: <%= 1 %>\n"
-      allow(database_config).to receive(:exist?).and_return(true)
-      allow(Pathname).to receive(:new).and_return(database_config)
+      @database_config = Tempfile.new
```

Comment:
We could remove the need for an instance variable here by using `around` rather than `before` and `after`

Reply:
rspec seems to get very unhappy when I try to use it.

Reply:
Seems strange since the next spec in this file uses it fine. Use that as a template and you should be okay.
If not, is there any more detail than "unhappy"? Maybe I can help.

Reply:
rspec will not permit an `allow` inside an `around` block:

```
allow(Pathname).to receive(:new).with(db_config_path).and_return(Pathname.new(@database_config.path))
       The use of doubles or partial doubles from rspec-mocks outside of the per-test lifecycle is not supported.
```

Reply:
Ah okay, well since this is a one spec context I would say put everything in the spec itself rather than using any hooks.

Reply:
The whole idea is to avoid the instance variable so using an `around` as well as a `before` doesn't really help.

Reply:
@carbonin Alright, got rid of the hooks. If this isn't what you want then you're gonna have to show me how it's done.